### PR TITLE
fix udev mouse

### PR DESCRIPTION
**Pull request by grant2258**
## Guidelines

1. Rebase before opening a pull request
2. If you are sending several unrelated fixes or features, use a branch and a separate pull request for each
3. If possible try squashing everything in a single commit. This is particularly beneficial in the case of feature merges since it allows easy bisecting when a problem arises

## Description

[Description of the pull request, detail any issues you are fixing or any features you are implementing]

## Related Issues

[Any issues this pull request may be addressing]

## Related Pull Requests

[Any other PRs from related repositories that might be needed for this pull request to work]

## Reviewers

[If possible @mention all the people that should review your pull request]

### CODE REVIEW

File: input/drivers/udev_input.c [hunk 1]
```diff
@@ -523,7 +523,7 @@ static bool udev_input_add_device(udev_input_t *udev,
       device->mouse.y_max = absinfo.maximum;
    }
    /* UDEV_INPUT_MOUSE may report in absolute coords too */
-   else if (type == UDEV_INPUT_MOUSE && ioctl(fd, EVIOCGABS(ABS_X), &absinfo) >= 0)
+   else if (type == UDEV_INPUT_MOUSE && ioctl(fd, EVIOCGABS(ABS_X), &absinfo) > 0)
```
[the remark this answers is not in the document]
Comment (grant2258) [reply input/drivers/udev_input.c]: the mouse works now there is a point where min  and = 0

[the remark this answers is not in the document]
Comment (Themaister) [reply input/drivers/udev_input.c]: I mean, the patch just check if ioctl is larger than 0 and not greater-equal to 0. So if anything, that should remove one possibility that a device is detected as a mouse.

[the remark this answers is not in the document]
Comment (grant2258) [reply input/drivers/udev_input.c]: https://github.com/libretro/RetroArch/blob/c7b3b0f7848f27333217cd66d239ba047db71617/input/drivers/udev_input.c#L528-L529
fails would you prefer i change it here

[the remark this answers is not in the document]
Comment (grant2258) [reply input/drivers/udev_input.c]: https://github.com/libretro/RetroArch/blob/07b17059f1bb98a29afa2033b4f74ce83acb5f5e/input/drivers/udev_input.c#L509-L513

i never even touched this one

[the remark this answers is not in the document]
Comment (Themaister) [reply input/drivers/udev_input.c]: oh, now I see, it checks ABS_X, not REL_X ...

[the remark this answers is not in the document]
Comment (Themaister) [reply input/drivers/udev_input.c]: I think a better approach will be to not error out and just set something sane like min = -1, max = 1 if the absinfo returns bogus data instead.

**Comment by joolswills** [comment]
Might be worth replacing PR template with some info about PR to aid those merging.

**Comment by grant2258** [comment]
ok let me post a bit of simplified code here 

  else if (type == UDEV_INPUT_MOUSE) 
   {
      if (ioctl(fd, EVIOCGABS(ABS_X), &absinfo) >= 0 ||  absinfo.minimum >= absinfo.maximum)
      {      
      	device->mouse.x_min = absinfo.minimum;
     	device->mouse.x_max = absinfo.maximum;
      }
all works fine.

 if you add an else condition at the end you will get (     else goto error; )

[INFO] [udev]: Keyboard #0 (/dev/input/event0).
[INFO] [udev]: Mouse #0 (/dev/input/event1).
[ERROR] [udev] Failed to open device: /dev/input/mouse0 (Inappropriate ioctl for device).

**Comment by grant2258** [comment]
@Autechre64 and @Themaister hopefully this code is good for you

**Comment by grant2258** [comment]
I would like to add the device is already open at this point we are just checking for abs. If that fails you should not be closing the device down.

**Comment by grant2258** [comment]
@joolswills i just tested this on retropie my logitech mouse and mini rii keyboard now work

File: input/drivers/udev_input.c [hunk 2]
```diff
@@ -505,38 +505,39 @@ static bool udev_input_add_device(udev_input_t *udev,
...
-
-      if (ioctl(fd, EVIOCGABS(ABS_Y), &absinfo) < 0 ||
-          absinfo.minimum >= absinfo.maximum)
-         goto error;
+      if (ioctl(fd, EVIOCGABS(ABS_X), &absinfo) >= 0)
+      {      
+	
+        if (absinfo.minimum >= absinfo.maximum )
+      	{
+          device->mouse.x_min = 0;
```
Comment (Themaister) [inline input/drivers/udev_input.c]: Should be -1 to 1 to avoid potential divide by zero scenario.

File: input/drivers/udev_input.c [hunk 3]
```diff
@@ -505,38 +505,39 @@ static bool udev_input_add_device(udev_input_t *udev,
...
-      device->mouse.y_max = absinfo.maximum;
+    if (ioctl(fd, EVIOCGABS(ABS_Y), &absinfo) >= 0)
+      {      
+	
+        if (absinfo.minimum >= absinfo.maximum )
+      	{
+          device->mouse.y_min = 0;
+          device->mouse.y_max = 0;
+        }
+	else
```
Comment (Themaister) [inline input/drivers/udev_input.c]: Formatting is wrong. Use 3 spaces for intendation.

Comment (grant2258) [reply input/drivers/udev_input.c]: i didnt realllise we had a format here good to know

File: input/drivers/udev_input.c (hunk 2 above)
Comment (grant2258) [reply input/drivers/udev_input.c]: which one do you wnt zero and which one do you want -1

**Pull request merged**